tests/e2e/Services/Databases/DatabasesCustomClientTest.php

```diff
@@ -890,4 +891,288 @@ public function testUpdateWithoutRelationPermission(): void
+                'number' => 100
+            ]
+        ]);
+
+        $this->assertEquals(401, $response['headers']['status-code']);
+
+        $collection = $this->client->call(Client::METHOD_PUT, '/databases/' . $data['databaseId'] . '/collections/' . $data['$id'], array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+            'x-appwrite-key' => $this->getProject()['apiKey']
+        ]), [
+            'name' => 'Bulk Updates Perms',
+            'documentSecurity' => true
+        ]);
+
+        $this->assertEquals(200, $collection['headers']['status-code']);
+
+        // TEST: Make sure we can update only documents we have permissions for
+        for ($i = 0; $i < 6; $i++) {
+            $doc = $this->client->call(Client::METHOD_POST, '/databases/' . $data['databaseId'] . '/collections/' . $data['$id'] . '/documents', array_merge([
+                'content-type' => 'application/json',
+                'x-appwrite-project' => $this->getProject()['$id'],
+            ], $this->getHeaders()), [
+                'documentId' => ID::unique(),
+                'data' => [
+                    'number' => $i,
+                ],
+                'permissions' => [
+                    Permission::update(Role::user($this->getUser()['$id']))
+                ]
+            ]);
+
+            $this->assertEquals(201, $doc['headers']['status-code']);
+        }
+
+        $doc = $this->client->call(Client::METHOD_POST, '/databases/' . $data['databaseId'] . '/collections/' . $data['$id'] . '/documents', [
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+            'x-appwrite-key' => $this->getProject()['apiKey']
+        ], [
+            'documentId' => ID::unique(),
+            'data' => [
+                'number' => 6,
+            ],
+            'permissions' => [
+                Permission::update(Role::user('user2'))
+            ]
+        ]);
+
+        $this->assertEquals(201, $doc['headers']['status-code']);
+
+        $response = $this->client->call(Client::METHOD_PATCH, '/databases/' . $data['databaseId'] . '/collections/' . $data['$id'] . '/documents', array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+        ], $this->getHeaders()), [
+            'data' => [
+                'number' => 100
+            ]
+        ]);
+
+        $this->assertEquals(200, $response['headers']['status-code']);
+        $this->assertEquals(6, $response['body']['modified']);
+
+        $documents = $this->client->call(Client::METHOD_GET, '/databases/' . $data['databaseId'] . '/collections/' . $data['$id'] . '/documents', array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+            'x-appwrite-key' => $this->getProject()['apiKey']
```

Comment:
Let's add read permission for the current user to all docs and use client auth here